astroquery/query.py

```diff
@@ -102,20 +107,30 @@ def hash(self):
         return self._hash
 
     def request_file(self, cache_location):
-        fn = os.path.join(cache_location, self.hash() + ".pickle")
+        fn = cache_location.joinpath(self.hash() + ".pickle")
         return fn
 
-    def from_cache(self, cache_location):
+    def from_cache(self, cache_location, cache_timeout):
         request_file = self.request_file(cache_location)
         try:
-            with open(request_file, "rb") as f:
-                response = pickle.load(f)
-            if not isinstance(response, requests.Response):
+            if cache_timeout is None:
+                expired = False
```

Comment:
If passing `None` as `cache_timeout` value is indeed an intended feature then it should be tested and documented. It doesn't necessarily have to be advertised in the narrative docs, but it should be explained in the docstring.